firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/TransformOperation.java

```diff
@@ -16,21 +16,34 @@
 
 import com.google.firebase.Timestamp;
 import com.google.firebase.firestore.model.value.FieldValue;
+import javax.annotation.Nullable;
 
 /** A transform within a TransformMutation. */
 public interface TransformOperation {
   /**
    * Computes the local transform result against the provided previousValue, optionally using the
    * provided localWriteTime.
    */
-  FieldValue applyToLocalView(FieldValue previousValue, Timestamp localWriteTime);
+  FieldValue applyToLocalView(@Nullable FieldValue previousValue, Timestamp localWriteTime);
 
   /**
    * Computes a final transform result after the transform has been acknowledged by the server,
    * potentially using the server-provided transformResult.
    */
-  FieldValue applyToRemoteDocument(FieldValue previousValue, FieldValue transformResult);
+  FieldValue applyToRemoteDocument(@Nullable FieldValue previousValue, FieldValue transformResult);
 
-  /** Returns whether this field transform is idempotent. */
-  boolean isIdempotent();
+  /**
+   * If applicable, returns the base value to persist with this transform. If a base value is
```

Comment:
"with this transform" is a tad misleading.  Maybe "for this transform"

Reply:
Good catch.